tests/host_reference/clover_reference.cpp

```diff
@@ -241,6 +241,68 @@ void twistCloverGamma5(void *out, void *in, void *clover, void *cInv, const int
   free(tmp1);
 }
 
+// Apply (A + i*mu*gamma_5*tau3 - epsilon*tau1) for QUDA_TWIST_GAMMA5_DIRECT
+// and   (A - i*mu*gamma_5*tau3 + epsilon*tau1)/(A^2 + mu^2 - epsilon^2) for QUDA_TWIST_GAMMA5_INVERSE
+void ndegTwistCloverGamma5(void *out1, void * out2, void *in1, void * in2,
+                           void *clover, void *cInv, const int dagger,
+                           const double kappa, const double mu,
+                           const double epsilon, const int parity,
+                           QudaTwistGamma5Type twist, QudaPrecision precision) {
+  void *tmp1 = malloc(Vh * spinor_site_size * precision);
+  void *tmp2 = malloc(Vh * spinor_site_size * precision);
+
+  double a = 0.0, b = 0.0;
+
+  if (twist == QUDA_TWIST_GAMMA5_DIRECT) {
+    a = 2.0 * kappa * mu;
+    b = -2.0 * kappa * epsilon;
+    
+    if (dagger) a *= -1.0;
+    
+    // apply_clover zeroes its output
+    apply_clover(tmp1, clover, in1, parity, precision);
+    apply_clover(tmp2, clover, in2, parity, precision);
+    // out = tmp + (i 2 kappa mu gamma_5 tau_3) * in
+    applyTwist(out1, in1, tmp1, a, precision);
+    applyTwist(out2, in2, tmp2, -a, precision);
+    // out += (-epsilon tau_1) * in 
+    axpy(b, in2, out1, Vh * spinor_site_size, precision);
+    axpy(b, in1, out2, Vh * spinor_site_size, precision);
+  } else if (twist == QUDA_TWIST_GAMMA5_INVERSE) {
+    void *tmptmp1 = malloc(Vh * spinor_site_size * precision);
+    void *tmptmp2 = malloc(Vh * spinor_site_size * precision);
+
+    a = -2.0 * kappa * mu;
```

Comment:
I don't understand yet, why the sign is different here.

Reply:
This is (C - i mu gamma5 tau 3 + eps tau1)/(C^2 + mu^2 - eps^2), whereas for `DIRECT`, we apply (C + i mu gamma5 tau3 - eps tau1).